cmd/watcher/main.go

```diff
@@ -66,6 +66,7 @@ var (
 	labelSelector           = flag.String("label_selector", "", "Selector (label query) to filter objects to be deleted. Matching objects must satisfy all labels requirements to be eligible for deletion")
 	requeueInterval         = flag.Duration("requeue_interval", 10*time.Minute, "How long the Watcher waits to reprocess keys on certain events (e.g. an object doesn't match the provided selectors)")
 	namespace               = flag.String("namespace", corev1.NamespaceAll, "Should the Watcher only watch a single namespace, then this value needs to be set to the namespace name otherwise leave it empty.")
+	checkOwner              = flag.Bool("check_owner", true, "If enabled, owner references will be checked while deleting objects")
```

Comment:
I thought we said that the default behavior would match the tekton pruner behavior (in which case the default should be `false`). Am I mistaken?

Reply:
I believe @sayan-biswas tried to preserve the existing behavior as per https://github.com/tektoncd/results/discussions/648#discussion-5776305, but I agree it makes more sense to change that to false.

Reply:
If we make it a default behaviour, then it would be a BREAKING change for the existing tekton results users. Kubernetes already implements the cascading deletion feature through owner references by default. IMO we should  keep that behaviour unless some users wants to override that, and this flag will help in that.

Reply:
Keeping backwards compatibility is important as well. After considering that again, I tend to agree with preserving the existing behavior. Users who want to change that, could use the new flag to do so.

Reply:
OK